code/game/atoms_movable.dm

```diff
@@ -139,21 +139,29 @@
 	if(destination)
 		if(pulledby)
 			pulledby.stop_pulling()
+
 		var/atom/oldloc = loc
-		var/same_loc = oldloc == destination.loc
+		var/same_loc = oldloc == destination

```

Comment:
Never should have been there https://github.com/tgstation/tgstation/pull/22490/files#diff-85b909bea63591ebec264bda09fbb017R145